### DISCUSSION TITLE
ReactJS app _kc.updateToken(minValidity 5). What is this minValidity?

### DISCUSSION THREAD

**Discussion by Paul6552** [keycloak/keycloak#16397] [Q&A]
Hello,

I try to configure my reactJS app with keycloak. I found this very good tutorial [Youtube Tutorial](https://www.youtube.com/watch?v=q50LxyGtEf0) and programmed it accordingly.

But in the video and also nowhere here on github or on the official homepage it is described why I need this minValidity.

In the source code I found this information:
`If the token expires within minValidity seconds, the token is refreshed.`

In my understanding, the number should be much larger than 5. For example, if the access token is still valid for 6 seconds, it will not be updated with `minValidity 5` and after 6 seconds the session is over and the user has to log in again.

How does this relate to the access token. Why is the value 5 by default and not some other value. Can someone please explain this to me. Thanks

**Comment by dsldiesel** [comment]
Same understanding here, it is not clear by docs!

**Comment by jonkoops** [comment]
> In my understanding, the number should be much larger than 5. For example, if the access token is still valid for 6 seconds, it will not be updated with minValidity 5 and after 6 seconds the session is over and the user has to log in again.

This is not the case, if the access token expires it can be refreshed by using the refresh token to obtain a new access token. This can be done by calling `updateToken()` on the Keycloak JS instance.

The reason why `minValidity` exists is because there might be a time difference between the server and the client, either because their internal clocks have drifted, or because a request might take longer than expected. By refreshing the access token earlier (before the `minValidity`), it is possible to avoid a situation where the token is sent and no longer valid.
